src/typing/env_js.ml

```diff
@@ -638,12 +748,12 @@ let merge_env =
     match env0, env1, env2 with
     | [], [], [] ->
       ()
-    | scope0 :: env0, scope1 :: env1, scope2 :: env2 -> Scope.(
+    | scope0 :: env0_, scope1 :: env1_, scope2 :: env2_ -> Scope.(
```

Comment:
:) I feel you on trying to avoid rebinding, but in this case I think it actually makes sense, given that you're peeling off the heads and (maybe) recursing straight into the tails, with nothing in between.